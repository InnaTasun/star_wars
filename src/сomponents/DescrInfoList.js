import ElemByUrl from './ElemByUrl';
import styled from 'styled-components';

function DescrInfoList(props) {
  const { dataArr, className } = props;
  let listItems;

  if (!dataArr || !dataArr.length) {
    listItems = 'n/a';
  } else {
    listItems = dataArr.map((itemUrl, index) => (
      <li className='info-list--info-item' key={index}>
        <ElemByUrl url={itemUrl} mainProp='name' addProp='model' />
        {index !== dataArr.length - 1 ? ', ' : ''}
      </li>
    ));
  }

  return <ul className={`${className} info-list`}>{listItems}</ul>;
}

const StyledDescrInfoList = styled(DescrInfoList)`
  display: flex;
  justify-content: start;
  flex-wrap: wrap;

  .info-list {
    &--info-item {
      list-style-type: none;
      margin-right: 0.5rem;
    }
  }
`;

export default StyledDescrInfoList;
