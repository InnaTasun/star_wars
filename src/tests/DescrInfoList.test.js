import React from 'react';
import { render, screen } from '@testing-library/react';
import DescrInfoList from '../сomponents/DescrInfoList';

const data = [
  'https://domain/vehicles/1/',
  'https://domain/vehicles/2/',
  'https://domain/vehicles/3/',
  'https://domain/vehicles/4/',
  'https://domain/vehicles/5/',
  'https://domain/vehicles/6/',
];

describe('DescrInfoList component', () => {
  test('renders with data', () => {
    render(<DescrInfoList dataArr={data} />);
    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.queryByRole('list').childElementCount).toBe(6);
  });

  test('renders without data', () => {
    render(<DescrInfoList />);
    expect(screen.getByRole('list').childElementCount).toBe(0);
  });

  test('renders with or without className', () => {
    let container = render(<DescrInfoList className='main-class' />).container;
    let firstDiv = container.firstElementChild;
    expect(firstDiv.className).toContain('main-class info-list');

    container = render(<DescrInfoList className='test-class' />).container;
    firstDiv = container.firstElementChild;
    expect(firstDiv.className).toContain('test-class info-list');

    container = render(<DescrInfoList />).container;
    firstDiv = container.firstElementChild;
    expect(firstDiv.className).not.toContain('main-class');
    expect(firstDiv.className).toContain('info-list');
  });
});

test("DescrInfoList snapshot", () => {
  const { container } = render(<DescrInfoList />);
  expect(container).toMatchSnapshot();
});