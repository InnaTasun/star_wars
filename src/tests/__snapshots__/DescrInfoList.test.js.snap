// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DescrInfoList snapshot 1`] = `
<div>
  <ul
    class="sc-gsDKAQ gLKjQI info-list"
  >
    n/a
  </ul>
</div>
`;
